doc/contribute/external.rst

```diff
@@ -0,0 +1,142 @@
+described in the following sections.
+
+Software License
+================
+
+.. note::
+
+   External source code licensed under the Apache-2.0 license is not subject to
+   this section.
+
+Integrating code into the Zephyr Project from other projects that use a license
+other than the Apache 2.0 license needs to be fully understood in
+context and approved by the `Zephyr governing board`_, as described in the
+`Zephyr project charter`_. The board will automatically reject licenses that have not been approved by the `Open Source
+Initiative (OSI)`_.
+
+.. _Zephyr governing board:
+   https://www.zephyrproject.org/governance/
+
+.. _Zephyr project charter:
+   https://www.zephyrproject.org/wp-content/uploads/sites/38/2020/09/CLEAN-LF-Zephyr-Charter-20200624-effective-20200901.pdf
+
+.. _Open Source Initiative (OSI):
+   https://opensource.org/licenses/alphabetical
+
+By carefully reviewing potential contributions and also enforcing a
+:ref:`DCO` for contributed code, we ensure that
+the Zephyr community can develop products with the Zephyr Project
+without concerns over patent or copyright issues.
+
+Merit
+=====
+
+Just like with any other regular contribution, one that contains external code
+needs to be evaluated for merit. However, in the particular case of code that
+comes from an existing project, there are additional questions that must be
+answered in order to accept the contribution.
+More specifically, the following will be considered by the Technical Steering
+Committee and evaluated carefully before the external source code is accepted
+into the project:
+
+- Is this the most optimal way to introduce the functionality to the project?
+  Both the cost of implementing this internally and the one incurred in
+  maintaining an externally developed codebase need to be evaluated.
+- Is the external project being actively maintained? This is particularly
+  important for source code that deals with security or cryptography.
+- Have alternatives to the particular implementation proposed been considered?
+  Are there other open source project that implement the same functionality?
+
+Mode of integration
+===================
+
+There are two ways of integrating external source code into the Zephyr Project,
+and careful consideration must be taken to choose the appropriate one for each
+particular case.
+
+Integration in the main tree
+----------------------------
+
+The first way to integrate external source code into the project is to simply
+import the source code files into the main ``zephyr`` repository. This
+automatically implies that the imported source code becomes part of the
+"mainline" codebase, which in turn requires that:
+
+- The code is formatted according to the Zephyr :ref:`coding style`
+- The code adheres to the project's :ref:`coding guidelines`
+- The code is subject to the same checks and verification requirements as the
+  rest of the code in the main tree, including static analysis
+
```

Comment:
Add an SPDX tag if not present.
Also add an entry here: https://docs.zephyrproject.org/latest/LICENSING.html#zephyr-licensing